src/xds/client.rs

```diff
@@ -299,26 +307,26 @@ impl AdsClient {
     }
 
     fn node(&self) -> Node {
-        let ip = std::env::var("INSTANCE_IP");
-        let ip = ip.as_deref().unwrap_or("1.1.1.1");
-        let pod_name = std::env::var("POD_NAME");
-        let pod_name = pod_name.as_deref().unwrap_or("");
-        let ns = std::env::var("POD_NAMESPACE");
-        let ns = ns.as_deref().unwrap_or("");
-        let node = std::env::var("NODE_NAME");
-        let node = node.as_deref().unwrap_or("");
+        let ip = std::env::var(INSTANCE_IP);
+        let ip = ip.as_deref().unwrap_or(DEFAULT_IP);
+        let pod_name = std::env::var( POD_NAME);
+        let pod_name = pod_name.as_deref().unwrap_or(EMPTY_STR);
+        let ns = std::env::var(POD_NAMESPACE);
+        let ns = ns.as_deref().unwrap_or(EMPTY_STR);
+        let node_name = std::env::var(NODE_NAME);
+        let node_name = node_name.as_deref().unwrap_or(EMPTY_STR);
         let mut metadata = Self::build_struct([
-            ("NAME", pod_name),
-            ("NAMESPACE", ns),
-            ("INSTANCE_IPS", ip),
-            ("NODE_NAME", node),
+            (POD_NAME, pod_name),
+            (POD_NAMESPACE, ns),
+            (INSTANCE_IP, ip),
+            (NODE_NAME, node_name),
         ]);
         metadata
             .fields
             .append(&mut Self::build_struct(self.config.proxy_metadata.clone()).fields);
 
         Node {
-            id: format!("ztunnel~{ip}~{pod_name}.{ns}~{ns}.svc.cluster.local"),
+            id: format!("ztunnel~{ip}~{pod_name}.{ns}.svc.cluster.local"),
```

Comment:
removed additional `{ns}` - Hope that is not intentional

Reply:
It is intended, align with sidecar

Reply:
got it